core/lib/libtomcrypt/sha512_accel.c

```diff
@@ -0,0 +1,183 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+#include <crypto/crypto_accel.h>
+#include <tomcrypt_private.h>
+
+/**
+   @param sha512.c
+   LTC_SHA512 by Tom St Denis
+*/
+
+#ifdef LTC_SHA512
+
+const struct ltc_hash_descriptor sha512_desc =
+{
+    "sha512",
+    5,
+    64,
+    128,
+
+    /* OID */
+   { 2, 16, 840, 1, 101, 3, 4, 2, 3,  },
+   9,
+
+    &sha512_init,
+    &sha512_process,
+    &sha512_done,
+    &sha512_test,
+    NULL
+};
+
+
+/* Implemented in assembly */
+int sha512_ce_transform(ulong64 *state, const unsigned char *buf, int blocks);
+
+static int sha512_compress_nblocks(hash_state *md, const unsigned char *buf,
+				   int blocks)
+{
+   void *state = md->sha512.state;
+
+   COMPILE_TIME_ASSERT(sizeof(md->sha512.state[0]) == sizeof(uint64_t));
+
+    crypto_accel_sha512_compress(state, buf, blocks);
+
+    return CRYPT_OK;
```

Comment:
One space too much?

Reply:
Right, it seems the indentation is 3 spaces. :-)